draft-ietf-dnsop-dns-catalog-zones.md

```diff
@@ -331,36 +373,49 @@ A catalog consumer SHOULD ignore custom properties it does not understand.
-Although they are regular DNS zones, catalog zones contain only information for
-the management of a set of authoritative nameservers.  For this reason,
-operators may want to limit the systems able to query these zones.  It may be
-inconvenient to serve some contents of catalog zones via DNS queries anyway due
-to the nature of their representation.  A separate method of querying entries
-inside the catalog zone may be made available by nameserver implementations
-(see (#implementationnotes)).
-
 Catalog updates should be automatic, i.e., when a nameserver that supports
 catalog zones completes a zone transfer for a catalog zone, it SHOULD apply
 changes to the catalog within the running nameserver automatically without any
 manual intervention.
 
-As with regular zones, primary and secondary nameservers for a catalog zone may
-be operated by different administrators.  The secondary nameservers may be
-configured as catalog consumer to synchronize catalog zones from the primary, but the primary's
-administrators may not have any administrative access to the secondaries.
-
 Nameservers MAY allow loading and transfer of broken zones with incorrect
-catalog zone syntax (as they are treated as regular zones), but catalog
-consumers MUST NOT process such broken zones as catalog zones.  For the purpose
-of catalog processing, the broken catalogs MUST be ignored.
+catalog zone syntax (as they are treated as regular zones).
+Catalog consumers MUST ignore broken catalog zones, however a clear error
+message stating the reason why the catalog zone is broken MUST be logged.
+
+When a previously correct catalog zone becomes a broken catalog zone, because
+of an update through an incremental transfer or otherwise, it loses its catalog
+meaning.
+No special processing occurs, no previously by this catalog configured member
+are removed or reconfigured in any way.
+
+If a name server restarts with a broken catalog zone, the broken catalog should
+not prevent the name server from starting up and serving the member zones in
+the last valid version of the catalog zone.
+
+Processing of a broken catalog will start (or resume) when the catalog turns
+into a correct catalog zone, for example by an additional update (through zone
+transfer or updates) fixing the catalog zone.
+
+Similarly, when a catalog zone expires, it loses its catalog meaning and
+MUST no longer be processed as such.
+No special processing occurs until the zone becomes fresh again.
+
+Great power comes with great responsibility: Catalog zones simplifies zone
+provisioning by orchestrating zones on secondary name servers from a single
+data source - the catalog.  Hence, the catalog source has great power and
+changes must be treated carefully. For example if the catalog is generated by
+some script and this script for whatever reason generates an empty catalog,
+millions of member zones may get deleted from their secondaries within seconds
+and all the affected domains may be offline in a blink.
```

Comment:
This "great power" warning would make more sense in the "Operational Notes" section to me.